api/v1/controllers/aspects.js

```diff
@@ -251,8 +297,44 @@ module.exports = {
    */
   deleteAspectWriter(req, res, next) {
     const userNameOrId = req.swagger.params.userNameOrId.value;
-    doDeleteOneAssoc(req, res, next, helper,
+    let aspectName;
+    let userName;
+    if (featureToggles.isFeatureEnabled('enableRedisSampleStore')) {
+      const params = req.swagger.params;
+      u.findByKey(helper, params)
+      .then((o) => u.isWritable(req, o,
+          featureToggles.isFeatureEnabled('enforceWritePermission')))
+      .then((o) => {
+        aspectName = o.name;
+        const options = {};
+        options.where = u.whereClauseForNameOrId(params.userNameOrId.value);
+        return u.findAssociatedInstances(helper,
+         params, helper.belongsToManyAssoc.users, options);
+      })
+      .then((o) => {
+        u.throwErrorForEmptyArray(o,
+           params.userNameOrId.value, userProps.modelName);
+
+        // the object "o" here is always an array of length 1
+        userName = o[0].dataValues.name;
+        return redisOps.getValue('aspect', aspectName);
+      })
+      .then((cachedAspect) => {
+        if (cachedAspect) {
+          cachedAspect.writers = cachedAspect.writers
+              .filter((writer) => writer !== userName);
+          return redisOps.hmSet('aspect', cachedAspect.name, cachedAspect);
+        }
+
+        return Promise.resolve(true);
+      })
+      .then(() => doDeleteOneAssoc(req, res, next, helper,
+        helper.belongsToManyAssoc.users, userNameOrId))
+      .catch((err) => u.handleError(next, err, helper.modelName));
+    } else {
```

Comment:
I just thought its better to add comments whenever we are adding a chunk of code, on why/what new code is. But you can ignore this if you feel this is unnecessary.